### DISCUSSION TITLE
pipx and shell enhancer

### DISCUSSION THREAD

**Discussion by Thadir** [pypa/pipx#1501] [Q&A] [answered]
Gooday,

I was somone can help me as I am tryingto figure out as brew has moved over to opholding the pep-0668. I tries installing some of my pyton comandline tools to pipx.

And [powerline-status](https://github.com/powerline/powerline) seems to work great. But its extendable with other libraries they call segments. Like for example [powerline-gitstatus](https://github.com/jaspernbrouwer/powerline-gitstatus).

But installing the plugin pipx is giving the following comment:

`
No apps associated with package powerline-gitstatus or its dependencies. If you are attempting to install a library, pipx should not be used. Consider using pip or a similar tool instead.`

How do I fix this.

**Comment by Gitznik** [comment, marked as answer]
I assume you probably want to inject the libraries into the main `powerline-status` installation.

The respective command for this is `pipx inject <application> <extension>`, so in your case `pipx inject powerline-status powerline-gitstatus`

**Reply by Thadir** [reply]
Quesstion where could I have red that in the manual I completly overlooked and worked as a charm

**Reply by Gitznik** [reply]
It is mentioned a few times within the docs (e.g. [here](https://pipx.pypa.io/stable/#inject-a-package) on the main page). But yeah, we get this question quite regularly. Maybe we can reword the docs slightly to make this feature easier to discover.

**Reply by Thadir** [reply]
I think it would be helpful to provide a hint after running pipx install <application>. This could be done by suggesting the appropriate command.

Additionally, in the case of an installation error, it would be beneficial to offer a hint, such as “Are you trying to do a extention maybe try?” and suggest the relevant command.

Many users expect pipx to be a drop-in replacement for pip, but it actually performs more complex operations under the hood. Including a link to the manual when unexpected behavior occurs could help clarify this and reduce the number of questions.
